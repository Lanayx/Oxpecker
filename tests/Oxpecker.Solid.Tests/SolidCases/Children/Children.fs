module Oxpecker.Solid.Tests.Cases.Children

open Oxpecker.Solid

[<SolidComponent>]
let Component (hello: string) (children: #HtmlElement) =
    h1() {
        hello
        children
    }

[<SolidComponent>]
let Test () =
    div() {
        Component "Hello1" <| br()
        Component "Hello2" <| Fragment() {
            i()
        }
        For(each = [|1..3|]) {
            yield fun i _ ->
                Component "Hello3" (Fragment() {
                    string i
                })
        }
    }
