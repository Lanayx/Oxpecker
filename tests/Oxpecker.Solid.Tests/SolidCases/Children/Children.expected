import { int32ToString } from "./fable_modules/fable-library-js.4.21.0/Util.js";
import { toArray } from "./fable_modules/fable-library-js.4.21.0/Seq.js";
import { rangeDouble } from "./fable_modules/fable-library-js.4.21.0/Range.js";

export function Component(hello, children) {
    return <h1>
        {hello}
        {children}
    </h1>;
}

export function Test() {
    return <div>
        {Component("Hello1", <br></br>)}
        {Component("Hello2", <>
            <i></i>
        </>)}
        <For each={toArray(rangeDouble(1, 1, 3))}>
            {(i, _arg) => Component("Hello3", <>
                {int32ToString(i)}
            </>)}
        </For>
    </div>;
}

